server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -126,6 +142,29 @@ class LivyServer extends Logging {
     sys.props("livy.server.serverUrl") = _serverUrl.get
   }
 
+  def runKinit(livyConf: LivyConf): Runnable = new Runnable {
+    override def run(): Unit = {
+      UserGroupInformation.getCurrentUser.reloginFromTicketCache()
+      val keytab = livyConf.get(LAUNCH_KERBEROS_KEYTAB)
+      val principal = livyConf.get(LAUNCH_KERBEROS_PRINCIPAL)
+      require(principal != null,
+        s"Kerberos requires ${LAUNCH_KERBEROS_KEYTAB.key} to be provided.")
+      require(keytab != null,
+        s"Kerberos requires ${LAUNCH_KERBEROS_PRINCIPAL.key} to be provided.")
+      val commands = ArrayBuffer.empty[String]
+      commands += "kinit"
+      commands += "-kt"
+      commands += keytab
+      commands += principal
```

Comment:
I am not sure whether we should introduce this option, as I feel it would make it more complicated to understand. '-l' can specify lifetime but it is constrained by max_life of KDC (if it is larger than max_life, then it won't take effect). For now, as long as user make the kinit refresh interval larger than ticket_lifetime, then the ticket will never expired